### PR TITLE
Unity upgrade to 9.x.x

### PR DESCRIPTION
**Pull request by RakshithaAcharya**
https://moengagetrial.atlassian.net/browse/MOEN-25155
Unity upgrade to 9.x.x

### CODE REVIEW

File: Core-SDK/Assets/MoEngage/Scripts/Model/InAppCampaignContext.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using UnityEngine;
+
+namespace MoEngage {
+
+  [System.Serializable]
+  public class InAppCampaignContext {
+    public string formattedCampaignId;
+    public Dictionary < string, object > attributes;
+
+    public InAppCampaignContext(Dictionary < string, object > campaignContextPayload) {
```
Comment (umangmoe) [inline Core-SDK/Assets/MoEngage/Scripts/Model/InAppCampaignContext.cs]: Move out the parsing logic to the parser class.

File: Core-SDK/Assets/MoEngage/Scripts/Model/SelfHandled.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+namespace MoEngage
+{
+  [System.Serializable]
+  public class SelfHandled
+  {
+    public string payload;
+    public long dismissInterval;
+    public bool isCancellable;
+
+    public SelfHandled(Dictionary < string, object > selfHandledDictionary) {
```
Comment (umangmoe) [inline Core-SDK/Assets/MoEngage/Scripts/Model/SelfHandled.cs]: move the parsing to parser class.

File: iOS/MoEngageConfiguration.h [hunk 3]
```diff
@@ -7,11 +7,42 @@
...
-#define kMoEngageLogsEnabled NO
-#define kDefaultIDFATrackingOptedOut NO
-#define kDefaultIDFVTrackingOptedOut NO
-#define kMoEngageUnityControllerSwizzlingEnabled YES
-#define kMoEngageEncryptNetworkRequests NO
+/// MoEngage Account Identifier
+#define kMoEngageAppID @"YOUR APPID"
+
+/// Set DataCenter
+#define kMoEngageRegion @"DATA_CENTER_01" // DATA_CENTER_01/DATA_CENTER_02/DATA_CENTER_03/DATA_CENTER_04/DATA_CENTER_05
```
Comment (umangmoe) [inline iOS/MoEngageConfiguration.h]: Let's not set a default data center if possible

**Review by umangmoe — CHANGES_REQUESTED** [review]

File: Core-SDK/Assets/MoEngage/Scripts/MoEngageUnityPlatform.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace MoEngage {
+  public interface MoEngageUnityPlatform {
```
Comment (umangmoe) [inline Core-SDK/Assets/MoEngage/Scripts/MoEngageUnityPlatform.cs:22]: add documentation

File: Core-SDK/Assets/MoEngage/Scripts/Utils/MoEUtils.cs [hunk 5]
```diff
@@ -0,0 +1,522 @@
...
+ *  Redistribution of source code or binaries is disallowed except with specific prior written permission. Any such redistribution must retain the above copyright notice, this list of conditions and the following disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using MoEMiniJSON;
+
```
Comment (umangmoe) [inline Core-SDK/Assets/MoEngage/Scripts/Utils/MoEUtils.cs:18]: All methods where we are converting string --> model or json --> model can be moved to a parser class

**Review by umangmoe — CHANGES_REQUESTED** [review]
- Add comments for all public models
- Properties object should not be in models, move it out.

File: Core-SDK/Assets/MoEngage/Scripts/Model/InAppCampaignContext.cs (hunk 1 above)
Comment (RakshithaAcharya) [reply Core-SDK/Assets/MoEngage/Scripts/Model/InAppCampaignContext.cs]: @umangmoe done

File: Core-SDK/Assets/MoEngage/Scripts/Model/SelfHandled.cs (hunk 2 above)
Comment (RakshithaAcharya) [reply Core-SDK/Assets/MoEngage/Scripts/Model/SelfHandled.cs]: @umangmoe done

File: iOS/MoEngageConfiguration.h (hunk 3 above)
Comment (RakshithaAcharya) [reply iOS/MoEngageConfiguration.h]: @umangmoe done

File: Core-SDK/Assets/MoEngage/Scripts/MoEngageUnityPlatform.cs (hunk 4 above)
Comment (RakshithaAcharya) [reply Core-SDK/Assets/MoEngage/Scripts/MoEngageUnityPlatform.cs:22]: @umangmoe done

File: Core-SDK/Assets/MoEngage/Scripts/Utils/MoEUtils.cs (hunk 5 above)
Comment (RakshithaAcharya) [reply Core-SDK/Assets/MoEngage/Scripts/Utils/MoEUtils.cs:18]: @umangmoe done

**Pull request merged**